src/main/java/org/opentripplanner/apis/transmodel/model/plan/ViaLocationInputType.java

```diff
@@ -1,78 +1,124 @@
+  public static final String FIELD_MINIMUM_WAIT_TIME = "minimumWaitTime";
+  public static final String FIELD_STOP_LOCATION_IDS = "stopLocationIds";
+
+  // TODO : Add coordinates
+  //private static final String FIELD_COORDINATES = "coordinates";
+  public static final String FIELD_VISIT = "visit";
+  public static final String FIELD_PASS_THROUGH = "passThrough";
+
+  private static final String DOC_LABEL =
+    "The label/name of the location. This is pass-through " +
+    "information and is not used in routing.";
+  private static final String DOC_MINIMUM_WAIT_TIME =
+    """
+    The minimum wait time is used to force the trip to stay the given duration at the
+    via-location before the trip is continued.
+    """;
+  private static final String DOC_STOP_LOCATION_IDS =
+    """
+    A list of stop locations. A stop location can be a quay, a stop place, a multimodal
+    stop place or a group of stop places. It is enough to visit ONE of the locations
+    listed.
+    """;
+
+  static final GraphQLInputObjectType VISIT_VIA_LOCATION_INPUT = GraphQLInputObjectType
+    .newInputObject()
+    .name(INPUT_VISIT_VIA_LOCATION)
+    .description(DOC_VISIT_VIA_LOCATION)
+    .field(b -> b.name(FIELD_LABEL).description(DOC_LABEL).type(GraphQLString))
+    .field(b ->
+      b
+        .name(FIELD_MINIMUM_WAIT_TIME)
+        .description(DOC_MINIMUM_WAIT_TIME)
+        .type(TransmodelScalars.DURATION_SCALAR)
+        .defaultValueLiteral(StringValue.of(Duration.ZERO.toString()))
+    )
+    .field(b ->
+      b
+        .name(FIELD_STOP_LOCATION_IDS)
+        .description(DOC_STOP_LOCATION_IDS)
+        .type(gqlListNonNullEntries(GraphQLString))
+    )
+    /*
+      TODO: Add support for coordinates
+       */
+    .build();
+
+  static final GraphQLInputObjectType PASS_THROUGH_VIA_LOCATION_INPUT = GraphQLInputObjectType
+    .newInputObject()
+    .name(INPUT_PASS_THROUGH_VIA_LOCATION)
+    .description(DOC_PASS_THROUGH_VIA_LOCATION)
+    .field(b -> b.name(FIELD_LABEL).description(DOC_LABEL).type(GraphQLString))
+    .field(b ->
+      // This is NOT nonNull, because we might add other parameters later, like 'list of line-ids'
+      b
+        .name(FIELD_STOP_LOCATION_IDS)
+        .description(DOC_STOP_LOCATION_IDS)
+        .type(gqlListNonNullEntries(GraphQLString))
+    )
+    .build();
+
+  public static final GraphQLInputObjectType VIA_LOCATION_INPUT = GraphQLInputObjectType
+    .newInputObject()
+    .name(INPUT_VIA_LOCATION)
+    .description(DOC_VIA_LOCATION)
+    .withDirective(OneOfDirective)
+    .field(b -> b.name(FIELD_VISIT).type(VISIT_VIA_LOCATION_INPUT))
+    .field(b -> b.name(FIELD_PASS_THROUGH).type(PASS_THROUGH_VIA_LOCATION_INPUT))
+    .build();
+
+  /**
+   * Return a {@link GraphQLList} with {@link GraphQLNonNull} elements.
+   */
+  private static GraphQLList gqlListNonNullEntries(GraphQLType wrappedType) {
+    return new GraphQLList(new GraphQLNonNull(wrappedType));
   }
```

Comment:
This was slightly repurposed and now it's called gqlListOfNonNullStrings, but it still feels like an utility method that could be put in some utils class.

Reply:
Yes, I did consider putting it in a shared location, but too infer it would be a massive change. I have my part of refactorings in this PR, but I need to stop at one point.